feature : 용어 사전 정리

mj950425 opened the pull request:
안녕하세요 리뷰어님~

이번 미션 페어인 @boradol 님과 함께 용어사전을 정리해봤습니다.

잘 부탁드리겠습니다.

mj19950425 committed:
feature : 용어 사전 정리

Gomding commented on README.md:
```diff
@@ -96,8 +96,93 @@ docker compose -p kitchenpos up -d
+|--------|-------------|-------------------------------|  
+| 가게 사장님 | store owner | 키친포스에서 음식을 판매하는 가게 사장님을 의미한다. |  
+| 고객     | customer    | 키친포스앱을 이용하는 고객을 의미한다.         |  
+| 배달 기사님 | rider       | 가게 사장님이 만든 음식을 배달하는 분을 의미한다.  |  
```
인물도 정리를 해주셨군요 👍 
대상이 누구인지 소통할 때 하나의 언어를 사용할 수 있겠어요! 💯

Gomding commented on README.md:
```diff
@@ -96,8 +96,93 @@ docker compose -p kitchenpos up -d
+| 고객     | customer    | 키친포스앱을 이용하는 고객을 의미한다.         |  
+| 배달 기사님 | rider       | 가게 사장님이 만든 음식을 배달하는 분을 의미한다.  |  
+
+### 외부 업체 및 시스템
```
해당 부분을 `외부 업체 및 시스템` 이라고 봐야할지 고민됐어요!
두 가지 고민이 있어요
1. 외부 업체와 시스템 카테고리를 분리해보는것은 어떻게 생각하시나요?
    - 처음 봤을 때 키친 포스도 외부 시스템? 이라는 착각을 잠깐 햇어요
2. 배달 대행 업체, 비속어 탐지 시스템의 경우 외부 업체일 수도 있지만 내부 자체 시스템이 될 수도 있을것같아요
    - 먼 미래의 생각일 수도 있지만요..  ㅎㅎ
    - 민준님은 어떻게 생각하시나요?

mj950425 replied:
리뷰 감사합니다~ 말씀하신것처럼 저도 헷갈릴 수 있다고 느껴지는데요. 

시스템이라는 카테고리로 수정해두는게 더 이해하기 쉬울 것 같아서 수정해두겠습니다!

Gomding commented on README.md:
```diff
@@ -96,8 +96,93 @@ docker compose -p kitchenpos up -d
+
+| 한글명      | 영문명                  | 설명             |  
+|----------|----------------------|----------------|  
+| 상품       | product              | 메뉴를 구성하는 단위이다. |  
```
`메뉴를 구성하는 단위이다.` 라는 설명이 자세해서 좋네요 👍 👍 👍

Gomding commented on README.md:
```diff
@@ -96,8 +96,93 @@ docker compose -p kitchenpos up -d
+
+| 한글명      | 영문명               | 설명                  |
+|----------|-------------------|---------------------|
+| 메뉴 그룹    | menu group        | 메뉴의 상위그룹을 의미한다.     |  
```
메뉴의 상위 그룹이 무엇인지 이해를 돕기위해 예시를 적어봐도 좋을것같아요!

> 메뉴의 상위그룹을 의미한다. 예시) 추천 메뉴, 두마리 치킨 메뉴, 세마리 치킨 메뉴

mj950425 replied:
넵 수정해두겠습니다!!

Gomding commented on README.md:
```diff
@@ -96,8 +96,93 @@ docker compose -p kitchenpos up -d
+| 메뉴 생성    | create menu    | 메뉴 그룹을 생성하는것을 의미한다.                   |  
+| 메뉴 노출 여부 | is displayed   | 메뉴가 고객에게 노출될 수 있는지 여부                 |
+| 메뉴 노출    | display menu   | 메뉴를 노출하는것을 의미한다.                      |  
+| 메뉴 숨김    | undisplay menu | 메뉴를 비노출하는것을 의미한다.                     |  
```
`노출 / 숨김` 에 대한 용어까지 꼼꼼하게 작성해주셨네요 👍

Gomding commented on README.md:
```diff
@@ -96,8 +96,93 @@ docker compose -p kitchenpos up -d
+| 한글명      | 영문명            | 설명                                    |  
+|----------|----------------|---------------------------------------|  
+| 메뉴       | menu           | 주문의 최소 단위를 의미한다. 한 개 이상의 상품으로 구성되어있다. |  
+| 메뉴 이름    | menu name      | 메뉴 그룹의 이름을 의미한다.                      |  
```
`메뉴 그룹의 이름을 의미한다.` 이 부분은 오타가 있는것 같네요 😎

mj950425 replied:
요것도 같이 수정해둘게요~

Gomding commented on README.md:
```diff
@@ -96,8 +96,93 @@ docker compose -p kitchenpos up -d
+
+| 한글명      | 영문명            | 설명                                    |  
+|----------|----------------|---------------------------------------|  
+| 메뉴       | menu           | 주문의 최소 단위를 의미한다. 한 개 이상의 상품으로 구성되어있다. |  
```
메뉴 그룹과 관련된 부분도 설명에 포함할 수 있겠어요!
민준님은 어떻게 생각하시나요?

mj950425 replied:
용어사전에서는 어느정도 추상화가 필요하다고 생각했습니다. 

그래서 너무 자세하지는않으면서, 헷갈리지않을정도의 매핑관계를 작성했는데요. 

메뉴그룹도 추가되면 좋을 것 같아서 수정해두겠습니다!

Gomding commented on README.md:
```diff
@@ -96,8 +96,93 @@ docker compose -p kitchenpos up -d
+|-----------------|---------------------------------------|-------------------------------|  
+| 주문 테이블          | order-table                           | 고객이 주문을 하기 위해 착석하는 테이블을 의미한다. |
+| 주문 테이블 명        | order-table name                      | 주문 테이블의 이름을 의미한다.             |
+| 점유여부            | occupied                              | 주문 테이블의 점유 여부를 의미한다.          |
```
점유 여부에 대한 각 테이블의 상태를 용어사전에 적어놓으면
소통할 때 도움이 될것같아요!

mj950425 replied:
반영해두겠습니다~

Gomding commented on README.md:
```diff
@@ -96,8 +96,93 @@ docker compose -p kitchenpos up -d
+| 주문 아이템 수량 | order line item quantity | 주문 받은 아이템의 수량을 의미한다.       |  
+| 주문 아이템 가격 | order line item price    | 주문 받은 아이템의 가격을 의미한다.       |  
+
+#### 주문 유형
```
주문 유형별로 주문 상태가 어떤 순서로 변경되는지 명시해주면 이해하는데 도움이 될것같아요~

> 배달 주문 : 승인 대기 > 승인 > 조리 완료 > 제공됨 > 배달 중 > 주문 배달됨

mj950425 replied:
이 부분이 조금 고민됐었는데요~ 용어 **사전**에서는 너무 자세한 부분까지는 다루지않고, 모델링 파트에서 다루려고했는데 어떻게 생각하시나요?

Gomding commented on README.md:
```diff
@@ -96,8 +96,93 @@ docker compose -p kitchenpos up -d
+| 포장      | takeout  | 음식이 포장되는 주문 유형을 의미한다.    |  
+| 매장 내 식사 | eat in   | 매장 내에서 식사하는 주문 유형을 의미한다. |
+
+#### 주문 상태
```
주문 상태에 몇 가지 누락된 것으로 보여요!

mj950425 replied:
수정해두겠습니다!

Gomding reviewed (changes requested):
안녕하세요 민준님!
이번 미션 함께하게된 박민영입니다 🤗
몇가지 코멘트 남겼으니 확인부탁드려요!

궁금한 점 있으면 언제든 DM이나 코멘트 남겨주세요~

mj19950425 committed:
fix : 용어 사전 정리

- 공통 생성
- 외부 업체 및 시스템을 시스템으로 변경
- 메뉴 그룹에 대한 예시 추가
- 메뉴 이름 오타 제거
- 주문 테이블 점유 여부에 대한 상태값 예시 추가. ex) 점유됨, 점유안됨
- 주문 상태 누락된 용어 추가

Gomding commented on README.md:
```diff
@@ -96,8 +96,100 @@ docker compose -p kitchenpos up -d
+| 주문 테이블 초기화      | clear order-table                     | 주문 테이블을 초기화한다.                         |
+| 주문 테이블 고객의 수 변경 | change order-table's number of guests | 생성된 주문 테이블의 고객의 수를 변경한다.               |
+
+### 주문
```
주문은 3가지 주문 유형(배달, 포장, 매장 내 식사) 를 포함하고 있는데
앞으로 서비스가 성장하고 요구사항이 추가되면서 용어도 많아질 거에요 :)
그런데 3개의 유형이 뭉쳐있으니 용어사전 유지보수도 어렵고 이해하는데도 점점 어려워질것같아요..

특히 주문이라는 공통점은 있지만 주문 유형에 따라 주문 상태에 액션도 달라질 거에요
예를들면 현재 `제공됨(Served)` 상태는 배달일때, 포장, 매장 내 식사에 따라 행위가 다를 수 있어요

이렇게 보면 주문을 세 가지 유형으로 분리해보는것도 좋을것 같은데 민준님은 어떻게 생각하시나요? 🤗

```
### 배달 주문

### 포장 주문

### 매장 내 식사 주문
```

Gomding reviewed (changes requested):
안녕하세요 민준님!
이미 용어사전 깔끔하게 정리해주셨는데
merge 전 마지막으로 의견 궁금한 부분이 있어서 코멘트 남겼어요 :)

궁금한 점 있으면 언제든 DM이나 코멘트 남겨주세요 🤗

mj19950425 committed:
fix : 용어 사전 정리

- 타입별 주문 대한 명시

Gomding commented on README.md:
```diff
@@ -96,8 +95,120 @@ docker compose -p kitchenpos up -d
+| 주문 배달됨 | delivered  | 배달주문일 때, 배달이 완료된 상태이다.                  |  
+| 완료     | completed  | 주문이 최종적으로 완료된 상태이다.                     |
+
+### 배달 주문
```
3가지 주문 유형으로 용어 사전을 분리해주셨군요 👍 

조금 더 나아가서 주문이라는 용어 그룹 자체를 없앤다는 생각으로
`배달 주문 / 포장 주문 / 매장 내 식사 주문`을 정리해보는건 어떻게 생각하시나요?

상태도 각각 정의 하구요!
이전 코멘트에서 설명드렸듯이 `Served` 는 각 주문 유형마다 의미하는 상태가 조금씩 다를거에요 😳

```
### 배달 주문
| 조리 완료 | Served | 조리가 완료되어 배달 기사가 가져갈 수 있는 상태 |

### 포장 주문
| 포장 완료 | Served | 조리가 완료되어 고객이 주문한 메뉴를 가져갈 수 있는 상태 |

### 매장 내 식사 주문
| 서빙 완료 | Served | 조리가 완료되어 손님에게 서빙이 완료된 상태 |
```

이처럼 같은 단어라도 어떤 주문이냐에 따라 의미가 달라질 수 있죠!
그렇다면 각 용어들이 중복되더라도(중복인듯 아닌듯..)
각각 정의하는것이 좋다고 생각해요 😁

mj950425 replied:
아하 저도 작성하면서 조금 이상하다는 느낌을 받았는데, 리뷰어님의 말씀대로 수정하는게 훨씬 깔끔할것 같네요. 다음 미션때 같이 반영해두겠습니다!

Gomding reviewed (approved):
안녕하세요 민준님!
한 가지 더 고민해주셨으면 하는 부분이 있는데
이번 단계는 merge 하고 다음 단계에서 고민해주시면 좋겠어요!
다음 단계도 화이팅입니다 💪
